src/NHibernate/Engine/ISessionImplementor.cs

```diff
@@ -187,6 +187,11 @@ public interface ISessionImplementor
 		void BeforeTransactionCompletion(ITransaction tx);
 
 		/// <summary>
+		/// 
+		/// </summary>
+		void FlushBeforeTransactionCompletion();
```

Comment:
Just wanted to stick with Hibernate. They also have it on the interface and only calling from inside the SessionImpl/StatelessSessionImpl